app/models/facility.rb

```diff
@@ -147,6 +147,10 @@ def source
     self
   end
 
+  def accessible_by(user, action: nil)
```

Comment:
A nil value for action will raise an error in `action_to_level`. Should we make it a required param instead?